https://pkg.go.dev/reflect#Value.SetIterKey missing

I am trying to build an app on tinygo.

It spits out this error:

```go
../../go/pkg/mod/github.com/fxamacker/cbor/v2@v2.7.0/encode_map.go:33:10: iterk.SetIterKey undefined (type *reflect.Value has no field or method SetIterKey)
../../go/pkg/mod/github.com/fxamacker/cbor/v2@v2.7.0/encode_map.go:34:10: iterv.SetIterValue undefined (type *reflect.Value has no field or method SetIterValue)
../../go/pkg/mod/github.com/fxamacker/cbor/v2@v2.7.0/encode_map.go:48:9: iterk.SetIterKey undefined (type *reflect.Value has no field or method SetIterKey)
../../go/pkg/mod/github.com/fxamacker/cbor/v2@v2.7.0/encode_map.go:49:9: iterv.SetIterValue undefined (type *reflect.Value has no field or method SetIterValue)
```

https://pkg.go.dev/reflect#Value.SetIterKey  and https://pkg.go.dev/reflect#Value.SetIterValue was added in Go1.18 and https://tinygo.org/docs/reference/lang-support/stdlib/ makes no mention of the function being missing.

How can I solve this?

Any a few more from standard library that is not mentioned in the documentation:

```go
# github.com/valyala/fasthttp
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/client.go:1919:9: c.Clone undefined (type *tls.Config has no field or method Clone)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/http.go:2264:13: w.ReadFrom undefined (type *os.File has no field or method ReadFrom)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/http.go:2278:13: w.ReadFrom undefined (type *net.TCPConn has no field or method ReadFrom)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/peripconn.go:50:7: undefined: tls.Conn
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/peripconn.go:58:31: undefined: tls.Conn
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1613:38: tlsConn.ConnectionState().NegotiatedProtocol undefined (type tls.ConnectionState has no field or method NegotiatedProtocol)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1646:14: undefined: os.Chmod
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1709:14: s.TLSConfig.BuildNameToCertificate undefined (type *tls.Config has no field or method BuildNameToCertificate)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1714:35: s.TLSConfig.Clone undefined (type *tls.Config has no field or method Clone)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1737:14: s.TLSConfig.BuildNameToCertificate undefined (type *tls.Config has no field or method BuildNameToCertificate)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1742:35: s.TLSConfig.Clone undefined (type *tls.Config has no field or method Clone)
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/server.go:1772:19: undefined: tls.X509KeyPair
../../go/pkg/mod/github.com/valyala/fasthttp@v1.59.0/tcpdialer.go:468:18: undefined: net.DefaultResolver
```

--- a/src/internal/reflectlite/value.go
+++ b/src/internal/reflectlite/value.go
@@ -1102,20 +1102,9 @@ func hashmapNewIterator() unsafe.Pointer
 func hashmapNext(m unsafe.Pointer, it unsafe.Pointer, key, value unsafe.Pointer) bool
 
 func (v Value) MapRange() *MapIter {
-	if v.Kind() != Map {
-		panic(&ValueError{Method: "MapRange", Kind: v.Kind()})
-	}
-
-	keyType := v.typecode.key()
-
-	keyTypeIsEmptyInterface := keyType.Kind() == Interface && keyType.NumMethod() == 0
-	shouldUnpackInterface := !keyTypeIsEmptyInterface && keyType.Kind() != String && !keyType.isBinary()
-
-	return &MapIter{
-		m:                  v,
-		it:                 hashmapNewIterator(),
-		unpackKeyInterface: shouldUnpackInterface,
-	}
+	iter := &MapIter{}
+	iter.Reset(v)
+	return iter
 }
 
 type MapIter struct {
@@ -1142,6 +1131,10 @@ func (it *MapIter) Key() Value {
 	return it.key.Elem()
 }
 
+func (v Value) SetIterKey(iter *MapIter) {
+	v.Set(iter.Key())
+}
+
 func (it *MapIter) Value() Value {
 	if !it.valid {
 		panic("reflect.MapIter.Value called on invalid iterator")
@@ -1150,12 +1143,33 @@ func (it *MapIter) Value() Value {
 	return it.val.Elem()
 }
 
+func (v Value) SetIterValue(iter *MapIter) {
+	v.Set(iter.Value())
+}
+
 func (it *MapIter) Next() bool {
 	it.key = New(it.m.typecode.Key())
 	it.val = New(it.m.typecode.Elem())
 
 	it.valid = hashmapNext(it.m.pointer(), it.it, it.key.value, it.val.value)
 	return it.valid
+}
+
+func (iter *MapIter) Reset(v Value) {
+	if v.Kind() != Map {
+		panic(&ValueError{Method: "MapRange", Kind: v.Kind()})
+	}
+
+	keyType := v.typecode.key()
+
+	keyTypeIsEmptyInterface := keyType.Kind() == Interface && keyType.NumMethod() == 0
+	shouldUnpackInterface := !keyTypeIsEmptyInterface && keyType.Kind() != String && !keyType.isBinary()
+
+	*iter = MapIter{
+		m:                  v,
+		it:                 hashmapNewIterator(),
+		unpackKeyInterface: shouldUnpackInterface,
+	}
 }
 
 func (v Value) Set(x Value) {

--- a/src/reflect/all_test.go
+++ b/src/reflect/all_test.go
@@ -381,8 +381,6 @@ func TestSetValue(t *testing.T) {
 	}
 }
 
-/*
-
 func TestMapIterSet(t *testing.T) {
 	m := make(map[string]any, len(valueTests))
 	for _, tt := range valueTests {
@@ -429,8 +427,6 @@ func TestMapIterSet(t *testing.T) {
 		t.Errorf("wanted %d alloc, got %d", want, got)
 	}
 }
-
-*/
 
 func TestCanIntUintFloatComplex(t *testing.T) {
 	type integer int
@@ -7955,6 +7951,8 @@ func TestConvertibleTo(t *testing.T) {
 	}
 }
 
+*/
+
 func TestSetIter(t *testing.T) {
 	data := map[string]int{
 		"foo": 1,
@@ -8043,6 +8041,8 @@ func TestSetIter(t *testing.T) {
 		})
 	}
 }
+
+/*
 
 func TestMethodCallValueCodePtr(t *testing.T) {
 	m := ValueOf(Point{}).Method(1)

--- a/src/reflect/value.go
+++ b/src/reflect/value.go
@@ -65,12 +65,24 @@ func (it *MapIter) Key() Value {
 	return Value{((*reflectlite.MapIter)(it)).Key()}
 }
 
+func (v Value) SetIterKey(iter *MapIter) {
+	v.Value.SetIterKey((*reflectlite.MapIter)(iter))
+}
+
 func (it *MapIter) Value() Value {
 	return Value{((*reflectlite.MapIter)(it)).Value()}
 }
 
+func (v Value) SetIterValue(iter *MapIter) {
+	v.Value.SetIterValue((*reflectlite.MapIter)(iter))
+}
+
 func (it *MapIter) Next() bool {
 	return ((*reflectlite.MapIter)(it)).Next()
+}
+
+func (iter *MapIter) Reset(v Value) {
+	(*reflectlite.MapIter)(iter).Reset(v.Value)
 }
 
 func (v Value) Set(x Value) {

